File: ConcretePractice/TestDrivenDevelopmentCodeFiles/CSharp/Solution/CalculatorTests/CalculatorTest.cs
using System;
using CalculatorLib;
using Xunit;

namespace CalculatorTests
{
    public class CalculatorTest
    {


        [Fact]
        public void SubtractionTest()
        {

        }
		[Fact]
	    public void subtractTest()
		{
			var calculator = new Calculator();
			int expected = 2;
			int actual = calculator.Subtract("5,3");
			Assert.Equal<int>(expected, actual);
		}

		[Fact]
	    public void subtract2Test()
		{
			var calculator = new Calculator();
			int expected = -8;
			int actual = calculator.Subtract("-5,3");
			Assert.Equal<int>(expected, actual);
		}

		[Fact]
	    public void subtract3Test()
		{
			var calculator = new Calculator();
			int expected = 2;
			int actual = calculator.Subtract("4,2");
			Assert.Equal<int>(expected, actual);
		}

		[Fact]
	    public void divide()
		{
			var calculator = new Calculator();
			int expected = 5;
			int actual = calculator.Divide("10,2");
			Assert.Equal<int>(expected, actual);
		}
		[Fact]
	    public void divide2()
		{
			var calculator = new Calculator();
			int expected = -5;
			int actual = calculator.Divide("10,-2");
			Assert.Equal<int>(expected, actual);
		}
		[Fact]
	    public void multiply()
		{
			var calculator = new Calculator();
			int expected = 50;
			int actual = calculator.Multiply("25,2");
			Assert.Equal<int>(expected, actual);
		}
		[Fact]
	    public void multiply2()
		{
			var calculator = new Calculator();
			int expected = 600;
			int actual = calculator.Multiply("150,4");
			Assert.Equal<int>(expected, actual);
		}

		
    	public void dividewithsomeexception()
		{
			var calculator = new Calculator();
            Assert.Throws<DivideByZeroException>(()=>calculator.Divide("10,0"));
		}

	}
}

File: 04_UnitTesting/Demos/Java/Example of Fresh Fixture/CalculatorProject/src/test/java/com/agile/calculator/CalculatorTest.java
package com.agile.calculator;

import static org.junit.Assert.assertEquals;

import org.junit.Before;
import org.junit.Test;

import com.agile.calculator.Calculator;

public class CalculatorTest {

	private Calculator calculator;

	@Test
	public void subtractTest() {
		calculator = new Calculator(); // Fresh Fixture
		int expected = 2;
		int actual = calculator.subtract("5,3");
		assertEquals(expected, actual);
	}

	@Test
	public void subtract2Test() {
		calculator = new Calculator(); // Fresh Fixture
		int expected = -8;
		int actual = calculator.subtract("-5,3");
		assertEquals(expected, actual);
	}

	@Test
	public void subtract3Test() {
		calculator = new Calculator(); // Fresh Fixture
		int expected = 2;
		int actual = calculator.subtract("4,2");
		assertEquals(expected, actual);
	}

	@Test
	public void divide() {
		calculator = new Calculator(); // Fresh Fixture
		int expected = 5;
		int actual = calculator.divide("10,2");
		assertEquals(expected, actual);
	}
	@Test
	public void divide2() {
		calculator = new Calculator(); // Fresh Fixture
		int expected = -5;
		int actual = calculator.divide("10,-2");
		assertEquals(expected, actual);
	}
	@Test
	public void multiply() {
		calculator = new Calculator(); // Fresh Fixture
		int expected = 50;
		int actual = calculator.multiply("25,2");
		assertEquals(expected, actual);
	}
	@Test
	public void multiply2() {
		calculator = new Calculator(); // Fresh Fixture
		int expected = 600;
		int actual = calculator.multiply("150,4");
		assertEquals(expected, actual);
	}
	
	@Test(expected = java.lang.ArithmeticException.class)
	public void dividewithsomeexception() {
		calculator = new Calculator(); // Fresh Fixture
		calculator.divide("10,0");
	}

}
